docs/reference/modules/cluster/disk_allocator.asciidoc

```diff
@@ -2,9 +2,60 @@
 ==== Disk-based shard allocation settings
 [[disk-based-shard-allocation-description]]
 // tag::disk-based-shard-allocation-description-tag[]
-{es} considers the available disk space on a node before deciding
-whether to allocate new shards to that node or to actively relocate shards away
-from that node.
+
+The <<shards-rebalancing-settings,balance>> of the cluster depends only on the
+number of shards on each node and the indices to which those shards belong. It
+considers neither the sizes of these shards nor the available disk space on
+each node, for the following reasons:
+
+* Disk usage changes over time. Balancing the disk usage of individual nodes
+would require a lot more shard movements, perhaps even wastefully undoing
+earlier movements. Moving a shard consumes resources such as I/O and network
+bandwidth and may evict data from the filesystem cache. These resources are
+better spent handling your searches and indexing where possible.
+
+* A cluster with accurately-balanced disk usage typically performs no better
+than one that has unequal disk usage across its nodes, as long as no disk is
+too full.
+
+The disk-based shard allocator ensures that all nodes have enough disk space
+without performing more shard movements than necessary. It allocates shards
+based on a pair of thresholds known as the _low watermark_ and the _high
+watermark_. Its primary goal is to ensure that no node breaches the high
+watermark, or at least that any such breach is only temporary. If a node
+breaches the high watermark then {es} will solve this by moving some of its
+shards onto other nodes in the cluster.
```

Comment:
> shards onto other nodes in the cluster.

Should we replace `cluster` for `data tier` here?

Reply:
This all applies whether you're using data tiers or not (e.g. you're running an OSS build). However I agree that we can mention data tiers somewhere, and IMO it fits better into the section lower down about allocation filtering rules.